src/gmail.d.ts

```diff
@@ -495,19 +501,19 @@ declare type GmailDomCompose = {
        options.type  string  to, cc, or bcc to check a specific one
        options.flat  boolean if true will just return an array of all recipients instead of splitting out into to, cc, and bcc
     */
-    recipients(options?: { type: string, flat: boolean }): GmailEmailAddress[];
+    recipients(options?: { type: 'to' | 'cc' | 'bcc', flat: boolean }): GmailDomComposeRecipients | string[];
```

Comment:
All changes up to this point looks good to me, as they're mostly documenting what we're doing. This is good.